File: NCPC/2017/ncpc2017-packages/judgingmoose/submissions/accepted/per.go
// I have no idea what I'm doing.

package main

import (
    "fmt"
)

func main() {
    var l, r, val int64

    fmt.Scanf("%d %d", &l, &r)

    if l > r {
       val = 2*l;
    } else {
       val = 2*r;
    }
    if val == 0 {
        fmt.Printf("Not a moose\n")
    } else if l == r {
        fmt.Printf("Even %d\n", val)
    } else {
        fmt.Printf("Odd %d\n", val)
    }

}

File: NCPC/2017/ncpc2017-packages/judgingmoose/submissions/accepted/per.hs
solve:: Int -> Int -> String
solve l r
   | l + r == 0 = "Not a moose"
   | l == r     = "Even " ++ show(2*l)
   | otherwise  = "Odd " ++ show(2 * (max l r))

parse :: String -> (Int, Int)
parse str = (vals !! 0, vals !! 1)
   where vals = (map read) $ words str

main = interact (uncurry solve . parse)
